daemon/daemon.go

```diff
@@ -17,17 +17,27 @@ type Daemon interface {
 }
 
 type daemon struct {
-	handler http.Handler
-	proxy   config.Proxy
+	handler     http.Handler
+	proxy       config.Proxy
+	healthCheck config.HealthCheck
 }
 
-func (d *daemon) Start() {
-	log.Printf("Listening on %s", d.proxy.Listen)
-	http.HandleFunc("/sphinx/health/check", func(rw http.ResponseWriter, req *http.Request) {
+// setUpHealthCheckService sets up a health check service at the given port
+// that can be pinged at the given endpoint to determine if Sphinx is still
+// running.
+func setUpHealthCheckService(port string, endpoint string) {
+	mux := http.NewServeMux()
+	mux.HandleFunc(endpoint, func(rw http.ResponseWriter, req *http.Request) {
 		rw.WriteHeader(http.StatusOK)
 	})
-	http.Handle("/", d)
-	log.Fatal(http.ListenAndServe(d.proxy.Listen, nil))
+	mux.Handle("/", http.NotFoundHandler())
+	go http.ListenAndServe(":"+port, mux)
+}
+
+func (d *daemon) Start() {
+	log.Printf("Listening on %s", d.proxy.Listen)
+	setUpHealthCheckService(d.healthCheck.Port, d.healthCheck.Endpoint)
```

Comment:
If the health check is optional, we should do some kind of check here and only call `setUpHealthCheckService` if the config specifies a health check?  I thought of this while wondering if we couldn't have a test that asserts the same - that we can successfully make a sphinx server without specifying a healthcheck in the config, so maybe we could add that test afterward?